vmware_guest_disk: Not specify filename for new disk

cuixiangz opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Not specify filename for new disk

if new disk filename is not specified, vmware esxi will set new disk
filename appending a not ever used number, so even recreate new disk with same controller number and same unit number, filename existence failure would not occur.

This change is to avoid of following issue.
```
msg:Failed to manage disks for virtual machine 'xxx' with exception : ('Cannot complete the operation because the file or folder /vmfs/volumes/yyy/xxx/xxx_1001_0.vmdk already exists'
```


<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
lib/ansible/modules/cloud/vmware/vmware_guest_disk.py
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
Add function test for this commit and test PASS.
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
First remove one existing disk without destroy file from datastore, recreate it with same controller number and unit number, recreate would success with filename appending new never used number.

Original disk information:
There are 6 disks, disk filenames are following, disk "3" is on controller number "0", unit number "4", filename is [LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_2.vmdk
"0"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/disk1.vmdk"
"1"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0.vmdk		"
"2"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_1.vmdk"
"3"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_2.vmdk"
   "controller_key": 1000
   "unit_number": 4
"4"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_3.vmdk"
"5"： "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_4.vmdk"

Step 1.  remove existing disk "3" on controller number '0' and unit number '4' without disk file from datastore.
```
- name: remove disks without destroy file
  vmware_guest_disk:
    hostname: "{{ vcenter_hostname }}"
    username: "{{ vcenter_username }}"
    password: "{{ vcenter_password }}"
    datacenter: "{{ dc1 }}"
    validate_certs: no
    name: "{{ virtual_machines[0].name }}"
    disk:
        - state: "absent"
          scsi_controller: 0
          unit_number: 4
          destroy: false
```

Step 2. recreate new disk on controller number '0' and unit number '4', recreate successfully, new filename is "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_5.vmdk", which is new and never used before.
```
            "5": {
                "backing_datastore": "LocalDS_0",
                "backing_disk_mode": "persistent",
                "backing_eagerlyscrub": true,
                "backing_filename": "[LocalDS_0] DC0_H0_VM0/DC0_H0_VM0_5.vmdk",
                "backing_thinprovisioned": null,
                "backing_writethrough": null,
                "capacity_in_bytes": null,
                "capacity_in_kb": 1048576,
                "controller_key": 1000,
...
                "key": 207,
                "label": "disk-1000-4",
...
                "unit_number": 4
            }

```

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

Tomorrow9 commented:
Hi @Akasurde, do you know is there any specific reason adding this here, there is no backing filename set in vmware_guest module? Thanks a lot.
```
if disk['filename'] is None:
      disk_spec.device.backing.fileName = "[%s] %s/%s_%s_%s.vmdk" % (
             disk['datastore'].name,
             vm_name, vm_name,
             str(scsi_controller),
             str(disk['disk_unit_number']))
```

cuixiangz committed:
Not specify new disk filename

If new disk filename is not specified, vmware esxi will set new disk
filename with a not ever used index, so even recreate new disk with
same controller number and same unit number, filename existence
failure would not occur.

cuixiangz commented:
@Tomorrow9 , I modified code to not set filename for new disk and added relevant test code. Test Pass. Thanks!

cuixiangz commented:
@Tomorrow9 , @Akasurde, if new disk filename is not specified, vmware esxi will set new disk
    filename with a not ever used index, so even recreate new disk with  same controller number and same unit number, filename existence. I pull a new commit without filename setting for new disk, and test PASS. Test Pass. Thanks!

goneri commented:
recheck

Tomorrow9 reviewed (approved):
LGTM, thanks for submitting this.

cuixiangz commented:
@goneri , @Akasurde，Could you help to review this PR? Thanks in advance.

goneri commented:
Hi!

Thank you @cuixiangz for your PR. Since March 2, 2020, the Ansible team does not accept any new changes on the VMware modules on the core `ansible/ansible` repository.
The development continue but in the new VMware collection repository: https://github.com/ansible-collections/vmware
Could you please reopen the PR there? Please also ping me directly, I will be happy to merge it.

Links:
- https://groups.google.com/forum/#!msg/ansible-devel/6UAfogEDtG8/wMTgxBGWCgAJ
- https://github.com/ansible-collections/overview/blob/master/README.rst

goneri commented:
I've manually migrated the PR in the new Git repository: https://github.com/ansible-collections/vmware/pull/14

I close this one sine it will never be merged.
